From patches Mon Sep 17 00:00:00 2001
From: "Jason J. Herne" <jjherne@us.ibm.com>
To: Anthony Liguori <aliguori@us.ibm.com>, jan.kiszka@siemens.com,
 mtosatti@redhat.com, agraf@suse.de, qemu-devel@nongnu.org,
 borntraeger@de.ibm.com, Jason Herne <jjherne@us.ibm.com>, R65777@freescale.com
Cc: Anthony Liguori <aliguori@us.ibm.com>, jan.kiszka@siemens.com,
 mtosatti@redhat.com, agraf@suse.de, qemu-devel@nongnu.org,
 Jason Herne <jjherne@us.ibm.com>, R65777@freescale.com
Date: Mon, 11 Mar 2013 13:58:56 -0400
Message-id: 1363024736-2650-3-git-send-email-jjherne@us.ibm.com
Subject: [PATCH 2/2] Regsync: Utilize selective runtime reg sync for hot code
 paths

Make use of new kvm_s390_get_registers_partial() for kvm_handle_css_inst() and
handle_hypercall() since they only need registers from the partial set and they
are called quite frequently.

Signed-off-by: Jason J. Herne <jjherne@us.ibm.com>
---
 target-s390x/kvm.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/target-s390x/kvm.c b/target-s390x/kvm.c
index 934757e..36861aa 100644
--- a/target-s390x/kvm.c
+++ b/target-s390x/kvm.c
@@ -468,12 +468,16 @@ static int kvm_handle_css_inst(S390CPU *cpu, struct kvm_run *run,
     int r = 0;
     int no_cc = 0;
     CPUS390XState *env = &cpu->env;
+    CPUState *cs = ENV_GET_CPU(env);
 
     if (ipa0 != 0xb2) {
         /* Not handled for now. */
         return -1;
     }
-    cpu_synchronize_state(env);
+
+    kvm_s390_get_registers_partial(cs);
+    cs->kvm_vcpu_dirty = true;
+
     switch (ipa1) {
     case PRIV_XSCH:
         r = ioinst_handle_xsch(env, env->regs[1]);
@@ -604,7 +608,10 @@ static int handle_priv(S390CPU *cpu, struct kvm_run *run,
 
 static int handle_hypercall(CPUS390XState *env, struct kvm_run *run)
 {
-    cpu_synchronize_state(env);
+    CPUState *cs = ENV_GET_CPU(env);
+
+    kvm_s390_get_registers_partial(cs);
+    cs->kvm_vcpu_dirty = true;
     env->regs[2] = s390_virtio_hypercall(env);
 
     return 0;
-- 
1.7.10.4

